src/firefly/js/drawingLayers/Catalog.js

```diff
@@ -142,11 +153,29 @@ function getLayerChanges(drawLayer, action) {
     var dd= Object.assign({},drawLayer.drawData);
     dd[DataTypes.HIGHLIGHT_DATA]= null;
     if (changes.tableData) dd[DataTypes.DATA]= null;
+    if (changes.selectInfo) dd[DataTypes.SELECTED_IDXS]= null;
     return Object.assign({}, changes,{drawData:dd});
 }
 
 
-const findColIdx= (columns,colId) => columns.findIndex( (c) => c.name===colId);
+
+function getDrawData(dataType, plotId, drawLayer, action, lastDataRet) {
+
+    const{tableData, columns}= drawLayer;
+    switch (dataType) {
+        case DataTypes.DATA:
+            return isEmpty(lastDataRet) ? computeDrawLayer(tableData, columns) : lastDataRet;
+        case DataTypes.HIGHLIGHT_DATA:
```

Comment:
If I have two catalogs, and I click on a point, the nearest points on both catalogs are "highlighted". I think there should be only one highlighted point, which corresponds to the active table.